src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -102,7 +102,13 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
           put0(next.key, next.value, next.hash, getOld = false)
         }
         this
-      case _ => super.addAll(xs)
+      case thatMap: Map[K, V] =>
+        thatMap.foreachEntry { (key: K, value: V) =>
+          put0(key, value, key.##, getOld = false)
```

Comment:
Here's the problem. You need `improveHash(key.##)`. The change comes from the commit titled "temp". It is not part of 2.12.x. Was this supposed to go into the separate PR for `foreachEntry`?

Reply:
Yes, looks like I got my workspaces confused when also working on https://github.com/scala/scala/pull/8761/files#diff-5be35908b1ec9bbb90c66342b71dd9afR106

Reply:
I've just added a new commit to revert `temp` because I'm never sure how to get git to DWIM when rebasing merges.